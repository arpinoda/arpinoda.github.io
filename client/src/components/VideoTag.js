import React from "react";
import PropTypes from "prop-types";

/**
 * Displays 'MOV' tag for Home Screen videos on non-touch devices.
 * @param {Object} props When props.isVisible controls tag visibility
 */
const VideoTag = props => {
  const { isVisible } = props;
  const style = {
    color: "rgba(0,0,0,0)",
    display: isVisible ? "block" : "none"
  };

  return <div style={style} className="spinner z2" />;
};

VideoTag.propTypes = {
  isVisible: PropTypes.bool
};

VideoTag.defaultProps = {
  isVisible: true
};

export default VideoTag;
