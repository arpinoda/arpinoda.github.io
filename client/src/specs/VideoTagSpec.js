import React from "react";
import renderer from "react-test-renderer";
import VideoTag from "../components/VideoTag";

describe("<VideoTag />", () => {
  test("Should not break with empty props", () => {
    const component = renderer.create(<VideoTag />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Should display tag", () => {
    const component = renderer.create(<VideoTag isVisible />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("Should NOT display tag", () => {
    const component = renderer.create(<VideoTag isVisible={false} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
