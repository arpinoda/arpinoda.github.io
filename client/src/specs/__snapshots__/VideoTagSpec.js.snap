// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoTag /> Should NOT display tag 1`] = `
<div
  className="spinner z2"
  style={
    Object {
      "color": "rgba(0,0,0,0)",
      "display": "none",
    }
  }
/>
`;

exports[`<VideoTag /> Should display tag 1`] = `
<div
  className="spinner z2"
  style={
    Object {
      "color": "rgba(0,0,0,0)",
      "display": "block",
    }
  }
/>
`;

exports[`<VideoTag /> Should not break with empty props 1`] = `
<div
  className="spinner z2"
  style={
    Object {
      "color": "rgba(0,0,0,0)",
      "display": "block",
    }
  }
/>
`;
